src/python_be.cc

```diff
@@ -1361,18 +1361,24 @@ ModelInstanceState::ProcessRequestsDecoupled(
   reporter.SetBatchStatistics(request_count);
 
   if (response_batch.data_->has_error) {
+    for (auto& infer_request : pb_infer_requests) {
+      // Reset the release flags for all the requests.
+      infer_request->SetReleaseFlags(TRITONSERVER_REQUEST_RELEASE_ALL);
+      // Clean up the response factory map.
+      {
+        std::lock_guard<std::mutex> guard{response_factory_map_mutex_};
```

Comment:
would it be better to hold the mutex across the whole operation? Seems odd at first glance to lock and unlock for each request -

Reply:
The mutex is only required for the operation on the map. Holding the mutex across the entire operation might slightly block other places where something is being inserted or read from the map.

Reply:
I think acquiring and releasing may have additional overhad in itself - but not positive